### ISSUE TITLE
Incorrect validation for IPv6-only subnet with dual-stack vpc

### ISSUE THREAD

**Issue by matsuu** [aws-cloudformation/cfn-lint#3963]
### CloudFormation Lint Version

1.25.1

### What operating system are you using?

Mac

### Describe the bug

When creating a subnet intended to be IPv6-only in a VPC that supports both IPv4 and IPv6, cfn-lint erroneously outputs the following:

```
E3014 Only one of ['CidrBlock', 'Ipv4IpamPoolId'] is a required property
```

In scenarios where the subnet is defined with `IPV6Native: true`, the properties `CidrBlock` and `Ipv4IpamPoolId` should not be required.  The current linting rule requires adjustments to accommodate this correct template.

cf. #3729 

### Expected behavior

no errors

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  VPC:
    Type: AWS::EC2::VPC
    Properties:
      CidrBlock: "10.0.0.0/16"

  Ipv6VPCCidrBlock:
    Type: AWS::EC2::VPCCidrBlock
    Properties:
      AmazonProvidedIpv6CidrBlock: true
      VpcId: !Ref VPC

  Ipv6OnlySubnet:
    DependsOn: Ipv6VPCCidrBlock
    Type: AWS::EC2::Subnet
    Properties:
      Ipv6Native: true
      Ipv6CidrBlock: !Select [0, !Cidr [ !Select [ 0, !GetAtt VPC.Ipv6CidrBlocks ], 1, 64]]
      VpcId: !Ref VPC
```
